otherlibs/win32unix/unix.ml

```diff
@@ -837,11 +842,19 @@ let make_process_env env =
     env;
   String.concat "\000" (Array.to_list env) ^ "\000"
 
+let create_process_aux prog args optenv fd1 fd2 fd3 =
+  let fd1 = dup ~cloexec:false fd1 in
+  let fd2 = dup ~cloexec:false fd2 in
+  let fd3 = dup ~cloexec:false fd3 in
+  let res = win_create_process prog args optenv fd1 fd2 fd3 in
+  close fd1; close fd2; close fd3;
+  res
```

Comment:
> I'm assuming dup

I don't think so.  Under Windows, handles are wrapped into head-allocated custom blocks, so dup can easily trigger the GC.